From c2eaaf39dc68e5d2acc4fbae95d815c85dd5a4fb
From: Carmelo Amoroso
Date: Mon, 13 Jun 2011 17:58:22 +0000
Subject: ldso_tls: fix compiler warning due to missing cast

Fix compiler warning (as below) due to missign cast

In file included from ldso/ldso/ldso.c:42:0:
ldso/ldso/dl-tls.c: In function 'init_tls':
ldso/ldso/dl-tls.c:1028:24: error: initialization makes pointer from integer without a cast

Signed-off-by: Carmelo Amoroso <carmelo.amoroso@st.com>
---
diff --git a/ldso/ldso/dl-tls.c b/ldso/ldso/dl-tls.c
index 43dd5a0..362efbb 100644
--- a/ldso/ldso/dl-tls.c
+++ b/ldso/ldso/dl-tls.c
@@ -1025,7 +1025,7 @@ init_tls (void)
 
 	/* And finally install it for the main thread.  If ld.so itself uses
 	   TLS we know the thread pointer was initialized earlier.  */
-	const char *lossage = TLS_INIT_TP (tcbp, USE___THREAD);
+	const char *lossage = (char *)TLS_INIT_TP (tcbp, USE___THREAD);
 	if(__builtin_expect (lossage != NULL, 0)) {
 		_dl_debug_early("cannot set up thread-local storage: %s\n", lossage);
 		_dl_exit(30);
--
cgit v0.9.0.1-2-gef13
